[ADD] product_pricelist_margins

chaule97 opened the pull request.

sebalix committed:
[ADD] New module 'product_main_supplierinfo'

oca-travis committed:
[UPD] Update product_main_supplierinfo.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[ADD] icon.png

Kev-Roche committed:
Added translation using Weblate (French)

OCA-git-bot committed:
[IMP] update dotfiles [ci skip]

Kev-Roche committed:
[FIX] product_stock_state product_template_view

TDu committed:
product_main_supplierinfo: fix default sellers for variant

If no specific record are found by the first filter. The Odoo default
method will return all active supplier info.
Which will include also supplier info record related to other specific
variant.

With this change if some supplier info related to the product
variant exist, they will be returned instead.

Also add a sort on the price to have the same behaviour than on a
purchase order line for the price selected.

OCA-git-bot committed:
product_main_supplierinfo 14.0.1.0.1

OCA-git-bot committed:
[UPD] README.rst

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (5 of 5 strings)

Translation: product-attribute-14.0/product-attribute-14.0-product_main_supplierinfo
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-14-0/product-attribute-14-0-product_main_supplierinfo/it/

ilyasProgrammer committed:
[ADD] product_internal_reference_generator

Commit:
[UPD] Update product_internal_reference_generator.pot

OCA-git-bot committed:
[BOT] post-merge updates

francesco-ooops committed:
Added translation using Weblate (Italian)

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (21 of 21 strings)

Translation: product-attribute-14.0/product-attribute-14.0-product_internal_reference_generator
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-14-0/product-attribute-14-0-product_internal_reference_generator/it/

ilyasProgrammer committed:
[IMP] product_internal_reference_generator: stock manager access

OCA-git-bot committed:
[BOT] post-merge updates

sbiosca-s73 committed:
[16.0][MIG]product_internal_reference_generator: Migration to 16.0

Ivorra78 committed:
Added translation using Weblate (Spanish)

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (5 of 5 strings)

Translation: product-attribute-14.0/product-attribute-14.0-product_main_supplierinfo
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-14-0/product-attribute-14-0-product_main_supplierinfo/es/

TDu committed:
Fix typo in product_main_supplierinfo

nguyenminhchien committed:
[ADD] product_pricelist_margin

OCA-git-bot committed:
[BOT] post-merge updates

legalsylvain committed:
[ADD] product_usability: New module that adds a new main entry menu for models present
in 'product' module. (product.product, product.category, etc...)

nguyenminhchien commented on product_pricelist_margins/models/product_pricelist_item.py:
```diff
@@ -0,0 +1,59 @@
+                )
+                and self.min_quantity
+            ):
+                res = item.product_id.taxes_id.compute_all(
```
product_id is not required and usually be empty. You should trust on `product_tmpl_id.taxes_id` than `product_id.taxes_id`

nguyenminhchien commented on product_pricelist_margins/models/product_pricelist_item.py:
```diff
@@ -0,0 +1,59 @@
+                res = item.product_id.taxes_id.compute_all(
+                    item.fixed_price,
+                    item.currency_id,
+                    item.min_quantity,
```
Indeed, self.min_quantity is usually zero. anw, don't need to input this argument (it will be 1 by default)

nguyenminhchien commented on product_pricelist_margins/models/product_pricelist_item.py:
```diff
@@ -0,0 +1,59 @@
+                    product=item.product_id,
+                )
+
+                price_vat_excl = res["total_excluded"] / item.min_quantity
```
```suggestion
                price_vat_excl = res["total_excluded"]
```
don't care item.min_quantity, from my previous comment, the price is computed by qty=1 by default

nguyenminhchien commented on product_pricelist_margins/models/product_pricelist_item.py:
```diff
@@ -0,0 +1,59 @@
+
+                price_vat_excl = res["total_excluded"] / item.min_quantity
+
+                cost = self.env.user.company_id.currency_id._convert(
```
it's more simple to call compute: `self.env.user.company_id.currency_id.compute(item.cost, item.currency_id)`

nguyenminhchien commented on product_pricelist_margins/models/product_pricelist_item.py:
```diff
@@ -0,0 +1,49 @@
+
+                margin = price_vat_excl - cost
+                percentage = (
+                    price_vat_excl and ((price_vat_excl - cost) / price_vat_excl) * 100
```
need to check price_vat_excl is NOT zero before dividing it.

legalsylvain committed:
[DOC] product_usability: add a note for V18 migration

legalsylvain committed:
[FIX] uom_category_active: CI is broken. because onchange doesn't raise an error. Swith from an onchange logic to an create / write logic to be sure to handle all the cases

OCA-git-bot committed:
Merge PR #1760 into 16.0

Signed-off-by simahawk

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #1757 into 16.0

Signed-off-by simahawk

Commit:
[UPD] Update product_stock_state.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: product-attribute-16.0/product-attribute-16.0-product_stock_state
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_stock_state/

samibc2c committed:
Added translation using Weblate (French)

samibc2c committed:
Translated using Weblate (French)

Currently translated at 28.5% (8 of 28 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_logistics_uom
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_logistics_uom/fr/

samibc2c committed:
Translated using Weblate (French)

Currently translated at 100.0% (9 of 9 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_dimension
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_dimension/fr/

OCA-git-bot committed:
Merge PR #1657 into 16.0

Signed-off-by legalsylvain

Commit:
[UPD] Update product_usability.pot

OCA-git-bot committed:
[BOT] post-merge updates

samibc2c committed:
Added translation using Weblate (French)

samibc2c committed:
Translated using Weblate (French)

Currently translated at 93.1% (27 of 29 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_assortment
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_assortment/fr/

samibc2c committed:
Added translation using Weblate (French)

samibc2c committed:
Translated using Weblate (French)

Currently translated at 71.4% (5 of 7 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_packaging_container_deposit
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_packaging_container_deposit/fr/

samibc2c committed:
Added translation using Weblate (French)

samibc2c committed:
Translated using Weblate (French)

Currently translated at 100.0% (1 of 1 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_total_weight_from_packaging
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_total_weight_from_packaging/fr/

samibc2c committed:
Added translation using Weblate (French)

samibc2c committed:
Translated using Weblate (French)

Currently translated at 90.0% (9 of 10 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_packaging_level_purchasable
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_packaging_level_purchasable/fr/

samibc2c committed:
Added translation using Weblate (French)

samibc2c committed:
Translated using Weblate (French)

Currently translated at 14.7% (5 of 34 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_packaging_level
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_packaging_level/fr/

samibc2c committed:
Added translation using Weblate (French)

samibc2c committed:
Translated using Weblate (French)

Currently translated at 100.0% (22 of 22 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_packaging_dimension
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_packaging_dimension/fr/

chaule97 committed:
[FIX]: Use cls.addClassCleanup(cls.loader.restore_registry)

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (16 of 16 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_usability
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_usability/it/

baguenth committed:
[FIX] product_uom_measure_type: Fixed the issue, that the module cannot be installed on migrated databases which already have a column 'measure_type'

OCA-git-bot committed:
Merge PR #1779 into 16.0

Signed-off-by legalsylvain

OCA-git-bot committed:
[BOT] post-merge updates

mmequignon committed:
[IMP] product_main_supplierinfo: pre-commit stuff

mmequignon committed:
[MIG] product_main_supplierinfo: Migration to 16.0

OCA-git-bot committed:
Merge PR #1485 into 16.0

Signed-off-by dreispt

Commit:
[UPD] Update product_internal_reference_generator.pot

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #1763 into 16.0

Signed-off-by dreispt

OCA-git-bot committed:
Merge PR #1786 into 16.0

Signed-off-by dreispt

Commit:
[UPD] Update product_main_supplierinfo.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: product-attribute-16.0/product-attribute-16.0-product_internal_reference_generator
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_internal_reference_generator/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: product-attribute-16.0/product-attribute-16.0-product_main_supplierinfo
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_main_supplierinfo/
